modules/sources/contracts/fetch-tokenpair-data.ts

```diff
@@ -193,6 +218,52 @@ function generateTokenPairs(filteredPools: PoolInput[]): TokenPair[] {
     return tokenPairs;
 }
 
+function generateBptTokenPairs(filteredPools: PoolInput[]): TokenPair[] {
```

Comment:
I would actually do that in the generateTokenPairs function, you should have all the info needed.